packages/components/src/mobile/dark-mode/index.native.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { eventEmitter, initialMode } from 'react-native-dark-mode';
+import React from 'react';
+
+export function useStyle( light, dark, theme ) {
+	const finalDark = {
+		...light,
+		...dark,
+	};
+
+	return theme === 'dark' ? finalDark : light;
+}
+
+// This function takes a component...
+export function withTheme( WrappedComponent ) {
+	return class extends React.Component {
+		constructor( props ) {
+			super( props );
+
+			this.state = {
+				mode: initialMode,
+			};
+		}
+
+		componentDidMount() {
+			eventEmitter.on( 'currentModeChanged', ( newMode ) => {
+				this.setState( { mode: newMode } );
+			} );
+		}
+
+		render() {
+			return <WrappedComponent theme={ this.state.mode } { ...this.props } />;
```

Comment:
I think your approach looks really good and clean. It does seem like the `theme` prop is only used as an argument to the `useStyle` function, so another option would be to just set the appropriate `useStyle` function as a prop itself. Then the `useStyle` function "prop" would only need to take the two styles as parameters, it wouldn't need to take the theme as a third parameter because it could access the `mode` state here directly. The benefit of this would be that it would avoid exposing the "theme" string to all the components that wrap this class. The negative is that the `withTheme` name fits _really_ well with your current implementation.

Obviously not a big deal, more just food for thought.

One other thing that I was wondering is if long-term the dark/light "theme" is something that we should put in the redux store. I'm honestly not sure, and I think your current approach is better at this time because it is both simpler and easier to change if/when we need to.

Reply:
Thank you for sharing your thoughts!

> another option would be to just set the appropriate useStyle function as a prop itself.

I like that idea. It's been quite annoying the need to pass the third argument all the time. But there has been a couple of cases where having the current theme has been helpful, cases where the color is given by html class name. But we still can pass both the function and the theme itself I guess :) 

I'll do some experiments about this. Thanks again!